patches/tModLoader/Terraria/ModLoader/GlobalBuff.cs

```diff
@@ -57,5 +58,28 @@ public abstract class GlobalBuff : ModType
 		/// </summary>
 		public virtual void DrawCustomBuffTip(string buffTip, SpriteBatch spriteBatch, int originX, int originY) {
 		}
+
+		/// <summary>
+		/// Allows you to draw things before the default draw code is ran. Return false to prevent drawing the buff. Returns true by default.
+		/// </summary>
+		/// <param name="spriteBatch">The spriteBatch that is drawn on</param>
+		/// <param name="type">The buff type</param>
+		/// <param name="buffIndex">The index in Main.LocalPlayer.buffType and .buffTime of the buff</param>
+		/// <param name="drawParams">The draw parameters for the buff</param>
+		/// <returns><see langword="true"/> for allowing drawing, <see langword="false"/> for preventing drawing</returns>
+		public virtual bool PreDraw(SpriteBatch spriteBatch, int type, int buffIndex, ref BuffDrawParams drawParams) {
+			return true;
+		}
+
+		/// <summary>
+		/// Allows you to draw things after the buff has been drawn. skipped is true if you or another mod has skipped drawing the buff (possibly hiding it or in favor of new visuals).
+		/// </summary>
+		/// <param name="skipped"><see langword="true"/> if you or another mod has skipped drawing the buff in PreDraw (possibly hiding it or in favor of new visuals)</param>
+		/// <param name="spriteBatch">The spriteBatch that is drawn on</param>
+		/// <param name="type">The buff type</param>
+		/// <param name="buffIndex">The index in Main.LocalPlayer.buffType and .buffTime of the buff</param>
+		/// <param name="drawParams">The draw parameters for the buff</param>
+		public virtual void PostDraw(bool skipped, SpriteBatch spriteBatch, int type, int buffIndex, BuffDrawParams drawParams) {
```

Comment:
`skipped` would be nicer as the last parameter here imo. Also, are there any use-cases for it? I don't think people would ever return false in PreDraw without doing their own drawing, they should disable an entire interface layer if they don't want buffs to be visible. And everyone should call tML hooks if they do buff rendering absolutely from scratch.

Reply:
One use case is to let the modder know something else returned false. Currently in all PostDraw hooks you cannot know this. Which sucks if you want to draw something that relies on the vanilla drawing, or your own PreDraw.

Though I have a hard time coming up with a convincing use case for `skipped`.